### PR TITLE
Adding Missing Incorporeal Checks

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Addresses certain problematic corner cases for the incorporeal status.

## Why It's Good For The Game

Addresses certain problematic corner cases for the incorporeal status.

Fixes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/6258

## Changelog
:cl:
fix: Incorporeal entities can no longer be attacked in melee, point blanked, grabbed, or shoved around by explosives.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/explosions.dm [hunk 1]
```diff
@@ -421,6 +421,11 @@ This way we'll be able to draw the explosion's expansion path without having to
 				var/atom/movable/thing_to_throw = am
 				if(thing_to_throw.anchored || thing_to_throw.move_resist == INFINITY)
 					continue
+				if(isliving(thing_to_throw))
+					var/mob/living/living_thing_to_throw = thing_to_throw
+					if(living_thing_to_throw.status_flags & INCORPOREAL) //Incorporeal things don't care about your insipid kinetics
+						continue
```
Comment (TiviPlus) [inline code/controllers/subsystem/explosions.dm:427]: Why not change move resist instead

File: code/game/objects/items/weapons/blades.dm [hunk 2]
```diff
@@ -128,9 +128,12 @@
 	return TRUE
 
 /obj/item/weapon/claymore/harvester/attack(mob/living/M, mob/living/user)
-	if(!loaded_reagent)
+	if(!loaded_reagent) //Can't attack the incorporeal
 		return ..()
```
Comment (TiviPlus) [inline code/game/objects/items/weapons/blades.dm:132]: uhh?

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 3]
```diff
@@ -145,6 +145,9 @@ Contains most of the procs that are called when a mob is attacked by something
 
 //Returns 1 if the attack hit, 0 if it missed.
 /mob/living/carbon/human/attacked_by(obj/item/I, mob/living/user, def_zone)
+	if(status_flags & INCORPOREAL) //We can't physically attack the incorporeal
+		return FALSE
+
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/human/human_defense.dm:150]: You shouldnt check this twice, either th object should check before attacking, or this code should happen

File: code/modules/mob/living/carbon/xenomorph/xenomorph.dm [hunk 4]
```diff
@@ -218,6 +218,8 @@
 	if(!Adjacent(AM)) //Logic!
 		return FALSE
 	var/mob/living/L = AM
+	if(status_flags & INCORPOREAL || L.status_flags & INCORPOREAL) //Incorporeal things can't grab or be grabbed.
+		return FALSE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenomorph.dm:222]: Status flags are on movable, you dont have to cast before

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/explosions.dm [hunk 5]
```diff
@@ -419,8 +419,9 @@ This way we'll be able to draw the explosion's expansion path without having to
 				continue
 			for(var/am in affected_turf)
 				var/atom/movable/thing_to_throw = am
-				if(thing_to_throw.anchored || thing_to_throw.move_resist == INFINITY)
+				if(thing_to_throw.anchored || thing_to_throw.move_resist == INFINITY || thing_to_throw.status_flags & INCORPOREAL)
```
Comment (TiviPlus) [inline code/controllers/subsystem/explosions.dm:422]: Again why do you not just set the wraiths move_resist to infinity so we can avoid this

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/explosions.dm (hunk 5 above)
Comment (Surrealaser) [reply code/controllers/subsystem/explosions.dm:422]: alright, fine, as long as it gets merged

Comment (Surrealaser) [reply code/controllers/subsystem/explosions.dm:422]: tbh, I prefer incorporeal as it future proofs but up to you

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**